File: 15_metadata_munging.r
library(data.table)
library(dplyr)

# export PATH="/well/lindgren/dpalmer/:$PATH"

# 1000 genomes population
dt_cell_pop <- fread(
  "/well/lindgren/UKBIOBANK/dpalmer/PRS_cell_data/superpopulation_assignments/superpopulation_labels.tsv") %>% 
  select(`sample.ID`, `classification_strict`, `super.population`)

# Cell-line ancestry assignment
directory <- "/well/lindgren/UKBIOBANK/dpalmer/PRS_cell_data/data/celldataB37/"
plink_cell <- paste0(directory, "celldataB37_dp_X")
system(paste("plink --bfile", paste0(directory, "celldataB37_dp"),
  "--out", plink_cell, "--geno 0.05 --make-bed"))
# Sample QC: call rate in cases or controls ≥ 0.97
system(paste("plink --bfile", plink_cell, "--out", plink_cell, "--mind 0.03 --make-bed"))
# SNP QC: call rate ≥ 0.98
system(paste("plink --bfile", plink_cell, "--chr 23 --out", plink_cell, "--geno 0.02 --make-bed"))

tmp_prune <- paste0(directory, "tmp")
plink_cell_pruned <- paste0(directory, "celldataB37_dp_X")
system(paste("plink", "--bfile", plink_cell, "--maf 0.01 --indep-pairwise 50 5 0.05 --out", tmp_prune))
system(paste("plink", "--bfile", plink_cell, "--extract", paste0(tmp_prune, ".prune.in"), "--make-bed --out", plink_cell_pruned))

# For ascertaining relatedness
plink_cell_auto <- paste0(directory, "celldataB37_dp_test-updated-autosomes-combined")
plink_cell_pruned_auto <- paste0(directory, "celldataB37_dp_test-updated-autosomes-combined_pruned")
system(paste("plink", "--bfile", plink_cell_auto, "--maf 0.01 --indep-pairwise 50 5 0.05 --out", tmp_prune))
system(paste("plink", "--bfile", plink_cell_auto, "--extract", paste0(tmp_prune, ".prune.in"), "--make-bed --out", plink_cell_pruned_auto))

File: 15_metadata_munging.py
import hail as hl
hl.init()

IMPUTESEX_FILE = '/well/lindgren/UKBIOBANK/dpalmer/PRS_cell_data/data/celldataB37/celldataB37_dp_impute_sex.tsv'
Y_NCALLED = '/well/lindgren/UKBIOBANK/dpalmer/PRS_cell_data/data/celldataB37/celldataB37_dp_y_called.tsv'

# Read in plink file and run sex imputation
X_PRUNED = '/well/lindgren/UKBIOBANK/dpalmer/PRS_cell_data/data/celldataB37/celldataB37_dp_X_pruned'
PLINK_CELL = '/well/lindgren/UKBIOBANK/dpalmer/PRS_cell_data/data/celldataB37/celldataB37_dp'

mt = hl.import_plink(
	bed = X_PRUNED + '.bed',
	bim = X_PRUNED + '.bim',
	fam = X_PRUNED + '.fam'
	)

mt_X = mt.filter_rows(mt.locus.in_x_nonpar())
imputed_sex = hl.impute_sex(mt_X.GT)
mt = mt.annotate_cols(impute_sex = imputed_sex[mt.s])

mt.cols().select('impute_sex').flatten().export(IMPUTESEX_FILE)

# Determine non-missing allele count on the y.
mt = hl.import_plink(
	bed = PLINK_CELL + '.bed',
	bim = PLINK_CELL + '.bim',
	fam = PLINK_CELL + '.fam'
	)

mt = mt.filter_rows(mt.locus.in_y_nonpar() | mt.locus.in_y_par())
mt = hl.sample_qc(mt, name='qc')

mt = mt.cols()
mt.select(n_called=mt.qc.n_called).export(Y_NCALLED)

# Also, evaluate the relatedness of the sequences in the cell-lines. Are the replicates really 100% related?
PLINK_CELL = '/well/lindgren/UKBIOBANK/dpalmer/PRS_cell_data/data/celldataB37/celldataB37_dp_test-updated-autosomes-combined_pruned'
mt = hl.import_plink(
	bed = PLINK_CELL + '.bed',
	bim = PLINK_CELL + '.bim',
	fam = PLINK_CELL + '.fam'
	)

IBD_OUTPUT = "/well/lindgren/UKBIOBANK/dpalmer/PRS_cell_data/data/celldataB37/celldataB37_ibd.tsv"
ibd_table = hl.identity_by_descent(mt)
ibd_table = ibd_table.flatten()
ibd_table.export(IBD_OUTPUT)
